documentation/design-docs/diagnostics-client-library.md

```diff
@@ -251,17 +342,14 @@ This is a class to represent an EventPipeSession. It is meant to be immutable an
 ```cs
 namespace Microsoft.Diagnostics.Client
 {
-    public class EventPipeSession
+    public class EventPipeSession : IDisposable
     {
-        public EventPipeSession(
-            int processId,
-            long sessionId,
-            IEnumerable<EventPipeProvider> providers,
-            int circularBufferMB,
-            bool rundownRequested
-        )
-
         public Stream EventStream { get; };
+
+        ///<summary>
+        /// Stops the given session
+        ///</summary>
+        public void Stop();
```

Comment:
Without a constructor or initialize method, how will `Stop()` know what session id to send?

Reply:
The constructor is internal. _sessionId is a private member